src/SurveyDesign.jl

```diff
@@ -157,4 +157,54 @@ struct ReplicateDesign <: AbstractSurveyDesign
     allprobs::Symbol # Right now only singlestage approx supported
     pps::Bool
     replicates::UInt
+    replicate_weights::Vector{Symbol}
+
+    # default constructor
+    function ReplicateDesign(
+        data::DataFrame,
+        cluster::Symbol,
+        popsize::Symbol,
+        sampsize::Symbol,
+        strata::Symbol,
+        weights::Symbol,
+        allprobs::Symbol,
+        pps::Bool,
+        replicates::UInt,
+        replicate_weights::Vector{Symbol}
+    )
+        new(data, cluster, popsize, sampsize, strata, weights, allprobs,
+           pps, replicates, replicate_weights)
+    end
+
+    # constructor with given replicate_weights
+    function ReplicateDesign(
+        data::AbstractDataFrame;
+        clusters::Union{Nothing,Symbol,Vector{Symbol}} = nothing,
+        strata::Union{Nothing,Symbol} = nothing,
+        popsize::Union{Nothing,Symbol} = nothing,
+        weights::Union{Nothing,Symbol} = nothing,
+        replicates::UInt,
+        replicate_weights::Vector{Symbol}
+    )
+        # call the SurveyDesign constructor
```

Comment:
So calling `SurveyDesign` to apply all common processing (at least for now). In case in future `ReplicateDesign` needs to diverge it still has capabilities?

Reply:
I think, since here we aren't doing much (just calling the constructor and returning the object with more fields), it'll be easy to change things if needed.